### DISCUSSION TITLE
Save 4 bands tiff image

### DISCUSSION THREAD

**Discussion by oldercoder** [libvips/libvips#4037] [Q&A]
Hi,
I have a 4 bands tiff remote sensing image, after processed in libvips, I use vips_tiffsave to save this image, but libvips always treat the 4th band as alpha band, is there any way to set VIPS_INTERPRETATION_MULTIBAND during save period?

any help is welcome, thanks.

**Comment by jcupitt** [comment]
Hello @oldercoder, yes, you probably have `interpretation` set to `srgb`. Just set `interpretation` before saving. Perhaps (untested):

```C
VipsImage *t = NULL;
if (vips_copy(image, &t, "interpretation", VIPS_INTERPRETATION_MULTIBAND, NULL) ||
    vips_tiffsave(image, "x.tif", NULL)) {
    VIPS_UNREF(t);
    return -1;
}
VIPS_UNREF(t);
```

The `copy` makes a new image with a different value for the interpretation tag, but pixels are copied with just a pointer copy, so there's no memory or CPU cost to consider.

**Reply by oldercoder** [reply]
Hi @jcupitt, thank you for your quick answer, however it's no luck... I used C++ api like below:
```
...
outImg = outImg.copy(VImage::option()
                                     ->set("interpretation", VIPS_INTERPRETATION_MULTIBAND)
                                     ->set("bands", 4));
outImg.tiffsave(outImgName.c_str(),
                      VImage::option()->set("pyramid", true)->
                              set("bigtiff", true)->
                              set("compression", VIPS_FOREIGN_TIFF_COMPRESSION_NONE));
```

any other suggestion?
